pkg/source/file/job.go

```diff
@@ -292,11 +292,19 @@ func (j *Job) File() *os.File {
 
 const tsLayout = "2006-01-02T15:04:05.000Z"
 
+func (j *Job) GetEncodeLineEnd() []byte {
+	return globalLineEnd.GetEncodeLineEnd(j.task.pipelineName, j.task.sourceName)
+}
+
+func (j *Job) GetLineEnd() []byte {
+	return globalLineEnd.GetLineEnd(j.task.pipelineName, j.task.sourceName)
+}
+
 func (j *Job) ProductEvent(endOffset int64, collectTime time.Time, body []byte) {
-	nextOffset := endOffset + 1
+	nextOffset := endOffset + int64(len(j.GetEncodeLineEnd()))
```

Comment:
`int64(len(j.GetEncodeLineEnd()))` can be initialized in the `newJob()` to achieve lock free. `Productevent()` is the core data link. We need to be very careful about performance